src/components/form/singUp/index.tsx

```diff
@@ -1,5 +1,66 @@
+import { FieldError } from 'react-hook-form';
+import FormInput from '../../common/formInput';
+import { SingUpFormHelper } from '../../../utils/formHelpers';
+import useSignUpForm from '../../../hooks/useSignUpForm';
+import { FormInputWrapper, StyledLoginButton } from './styled';
+
+export interface ISignUp {
+  email?: string;
+  password?: string;
+  passwordConfirmation?: string;
+}
+export interface FieldErrors {
+  [key: string]: FieldError | undefined;
+}
 const SignUpForm = () => {
-  return <div>SignUpForm</div>;
+  const {
+    showPassword,
+    handleClickShowPassword,
+    handleMouseDownPassword,
+    handleSubmit,
+    register,
+    watch,
+    errors,
+    onSubmit,
+    generateAdornmentProps,
+  } = useSignUpForm();
+
+  const password = watch('password', '');
+  const passwordConfirmation = watch('passwordConfirmation', '');
+
+  return (
+    <FormInputWrapper>
+      <form onSubmit={handleSubmit(onSubmit)}>
+        {SingUpFormHelper.map(({ name, label, validations, placeholder, type }) => (
+          <FormInput
+            key={name}
+            name={name}
+            label={label}
+            validations={validations}
+            placeholder={placeholder}
+            type={type}
+            showPassword={showPassword}
```

Comment:
It is not a reusable component anymore
FormInput knows a lot of unnecessary info.
password, showPassord, handleClickShowPassword, handleMouseDownPassword, passwordConfirmation
To many specific props

Reply:
I assume I am supposed to return it to reusable component again but I am not sure how exactly to do that.